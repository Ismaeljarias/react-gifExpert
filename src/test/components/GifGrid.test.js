import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("GifGrid Tests", () => {
  const category = "One Punch";

  test("should show <GifGrid /> correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when images are loaded useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/img.jpg",
        title: "Title",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);

    expect(wrapper.find("GridGifItem").length).toBe(gifs.length);
  });
});
