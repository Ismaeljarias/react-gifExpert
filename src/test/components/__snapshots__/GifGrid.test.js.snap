// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Tests should show <GifGrid /> correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid Tests should show items when images are loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GridGifItem
      id="ABC"
      key="ABC"
      title="Title"
      url="https://localhost/img.jpg"
    />
  </div>
</Fragment>
`;
